src/app/transactions/page.tsx

```diff
@@ -1,19 +1,124 @@
+  | 'tx_status'
+  | 'block_height'
+  | 'fee_rate'
+  | 'block_time'
+  | 'block_height'
+  | 'sender_address'
+> & {
+  token_transfer?: Pick<
+    NonNullable<TokenTransferTransaction['token_transfer']>,
+    'amount' | 'recipient_address'
+  >;
+  smart_contract?: Pick<NonNullable<SmartContractTransaction['smart_contract']>, 'contract_id'>;
+  contract_call?: Pick<
+    NonNullable<ContractCallTransaction['contract_call']>,
+    'contract_id' | 'function_name'
+  >;
+  coinbase_payload?: Pick<NonNullable<CoinbaseTransaction['coinbase_payload']>, 'alt_recipient'>;
+  tenure_change_payload?: Pick<
+    NonNullable<TenureChangeTransaction['tenure_change_payload']>,
+    'cause'
+  >;
+};
+
+function compressTransactions(transactions: Transaction[]): CompressedTxTableData[] {
+  return transactions.map(tx => {
+    const minimalTx: CompressedTxTableData = {
+      tx_id: tx.tx_id,
+      tx_type: tx.tx_type,
+      tx_status: tx.tx_status,
+      block_height: tx.block_height,
+      fee_rate: tx.fee_rate,
+      block_time: tx.block_time,
+      sender_address: tx.sender_address,
+    };
+
+    if ('token_transfer' in tx && tx.token_transfer) {
+      minimalTx.token_transfer = {
+        amount: tx.token_transfer.amount,
+        recipient_address: tx.token_transfer.recipient_address,
+      };
+    }
+
+    if ('smart_contract' in tx && tx.smart_contract) {
+      minimalTx.smart_contract = {
+        contract_id: tx.smart_contract.contract_id,
+      };
+    }
+
+    if ('contract_call' in tx && tx.contract_call) {
+      minimalTx.contract_call = {
+        contract_id: tx.contract_call.contract_id,
+        function_name: tx.contract_call.function_name,
+      };
+    }
+
+    if ('coinbase_payload' in tx && tx.coinbase_payload) {
+      minimalTx.coinbase_payload = {
+        alt_recipient: tx.coinbase_payload.alt_recipient,
+      };
+    }
+
+    if ('tenure_change_payload' in tx && tx.tenure_change_payload) {
+      minimalTx.tenure_change_payload = {
+        cause: tx.tenure_change_payload.cause,
+      };
+    }
+
+    return minimalTx;
+  });
 }
 
-export default async function (props: { searchParams: Promise<TxPageFilters> }) {
-  const searchParams = await props.searchParams;
 
-  const { startTime, endTime, fromAddress, toAddress } = searchParams;
+export default async function ({
+  searchParams: { startTime, endTime, fromAddress, toAddress, chain, api },
+}: {
+  searchParams: TxPageFilters;
+}) {
+  const params = new URLSearchParams({
+    limit: `${TX_TABLE_PAGE_SIZE}`,
+    offset: '0',
+    ...(startTime && { start_time: startTime }),
+    ...(endTime && { end_time: endTime }),
+    ...(fromAddress && { sender_address: fromAddress }),
+    ...(toAddress && { recipient_address: toAddress }),
+  });
+  const apiUrl = api ? api : getApiUrl(chain || 'mainnet');
+  const response = await fetch(`${apiUrl}/extended/v1/tx/?${params.toString()}`);
+  const data = await response.json();
+  const compressedData = {
+    ...data,
```

Comment:
It looks like full data is still being passed to client.

Reply:
I am not passing the full data. You can see that I am overwriting the results with the results that I have compressed.

Reply:
image.png

Reply:
![image](https://github.com/user-attachments/assets/e2ef73fe-b3c7-41a5-8b58-42377ea09fac)